### PR TITLE
Add CI for MTU migration

### PR DESCRIPTION
**Pull request by jcaamano**
Add CI for MTU migration, new network feature for 4.10.

Signed-off-by: Jaime Caamaño Ruiz <jcaamano@redhat.com>

### CODE REVIEW

**Comment by jcaamano** [comment]
WIP

/hold

File: ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-master.yaml
Comment (jluhrsen) [inline ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-master.yaml:242]: is this just doing a decrease, then increase, then running tests?

do you want to run tests before or in the middle of the decrease/increase?

Comment (jcaamano) [reply ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-master.yaml:242]: Yes.

We can only run tests once, so doing that at the end.

Comment (jluhrsen) [reply ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-master.yaml:239]: ok. is it important to run e2e after having decreased MTU as well? if so, we'll need a second job I guess.

Comment (jcaamano) [reply ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-master.yaml:239]: It might provide some limited value, but then we would need 3 jobs more for a total of 6 if we want to cover everything. The current setup is what I thought it would be reasonable to have initially.

Comment (jluhrsen) [reply ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-master.yaml:239]: sure, whatever seems right to you.

**Comment by jcaamano** [comment]
@jluhrsen I think we can go ahead with this

/hold cancel

File: ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-master.yaml
Comment (jluhrsen) [inline ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-master.yaml:199]: don't you need the DEV_SCRIPTS config here to use NETWORK_TYPE=OVNKubnernetes ?

File: ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh
Comment (jluhrsen) [inline ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh:137]: nit: echo something here indicating we are going to bail on the job because we these alerts.

File: ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh
Comment (jluhrsen) [inline ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh:341]: nothing to poll on instead?

File: ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh
Comment (jluhrsen) [inline ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh:360]: maybe I missed it, but are we validating the correct MTU is there after the patch is done to change it? Also, wondering
how important it might be to check directly on the hosts (e.g., ip link cmd or whatever)

File: ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-master.yaml
Comment (jcaamano) [reply ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-master.yaml:199]: No, in this case OVN config is handled through task `ovn-conf` within the `openshift-e2e-gcp-ovn` workflow. 

I will set them all as optional.

File: ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh
Comment (jcaamano) [reply ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh:137]: Ok on this one here.

On the one below we are already doing `echo "Failed to get default interface"`. Is it something else?

File: ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh
Comment (jcaamano) [reply ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh:341]: nothing that we know of for this specific thing. But then we do poll on `wait_for_final_mc`. The consequence though if we don't space out these updates, `wait_for_final_mc` waits 10 minutes or more. If we do space them it takes a lot of less time. But is not a job-will-fail consequence.

File: ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh
Comment (jcaamano) [reply ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh:360]: Not easy to do a full check on this without getting into implementation details and/or excessive complications. I will try to bake something in.

Comment (jcaamano) [reply ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh:360]: Just to clarify, there some other things that would surely fail during the process and that would be detected anyways.

File: ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-master.yaml
Comment (jluhrsen) [reply ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-master.yaml:200]: gotcha. I saw it below in the e2e-network-mtu-migration-ovn-ipv6 job which made me think
it was missing here. But, now I realize you could also do this same thing below I think and not
have to include DEVSCRIPTS_CONFIG if you use the baremetalds-e2e-ovn-ipv6 workflow.

NBD though. either works fine.

File: ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh
Comment (jluhrsen) [reply ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh:137]: I'm just used to digging through CI logs and sometimes it's not clear where a "FAIL" thing is just printed as
a log, or when that failure actually led to the exit 1 which is why the job is marked as a failure. so any verbosity
around that can help whoever is debugging later.

File: ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh
Comment (jluhrsen) [reply ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh:369]: ok, makes sense.

File: ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh
Comment (jluhrsen) [reply ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh:388]: ok. my thought was that we could end up in a state where we set the MTU, and then read it with some oc command and
it gives back what we expect, but then in reality underneath it's not really changed. but, maybe those "some other things"
that would surely fail if they are not set like we expect is good enough to validate it. your call.

File: ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-master.yaml
Comment (jcaamano) [reply ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-master.yaml:200]: Ok, I missed that one. I will try it out.

File: ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh
Comment (jcaamano) [reply ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh:137]: got it, I will qualify all errors with `Error`.

File: ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh
Comment (jcaamano) [reply ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh:415]: Added some checks for node and pod MTU.

File: ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh
Comment (jcaamano) [reply ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh]: It's dues this bug https://bugzilla.redhat.com/show_bug.cgi?id=2005694

Comment (jluhrsen) [reply ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh]: good find. maybe add this bug in the comment so someone might notice it later and be able to remove
the added delays when the bug is fixed.

File: ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-master.yaml
Comment (jluhrsen) [reply ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-master.yaml:200]: 👀

Comment (jluhrsen) [reply ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-master.yaml:200]: what examples do you have? looking at this recent [one](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/24734/rehearse-24734-pull-ci-openshift-cluster-network-operator-master-e2e-network-mtu-migration-ovn-ipv4/1509163839768236032), it looks like the must-gather step (after the tests)
failed because of some imagePull issue.

**Comment by jluhrsen** [comment]
looks like the tests were running, but maybe the job timeout needs to be increased? aborted after the 4h timeout:

```
INFO[2022-08-17T12:16:04Z] Running step e2e-network-mtu-migration-ovn-ipv6-baremetalds-e2e-test. 
{"component":"entrypoint","file":"k8s.io/test-infra/prow/entrypoint/run.go:165","func":"k8s.io/test-infra/prow/entrypoint.Options.ExecuteProcess","level":"error","msg":"Process did not finish before 4h0m0s timeout","severity":"error","time":"2022-08-17T13:05:45Z"}

```

**Comment by jcaamano** [comment]
> looks like the tests were running, but maybe the job timeout needs to be increased? aborted after the 4h timeout:
> 
> ```
> INFO[2022-08-17T12:16:04Z] Running step e2e-network-mtu-migration-ovn-ipv6-baremetalds-e2e-test. 
> {"component":"entrypoint","file":"k8s.io/test-infra/prow/entrypoint/run.go:165","func":"k8s.io/test-infra/prow/entrypoint.Options.ExecuteProcess","level":"error","msg":"Process did not finish before 4h0m0s timeout","severity":"error","time":"2022-08-17T13:05:45Z"}
> ```

Is that something we can do per job. I though it was a hard timeout  of 4h for all jobs.

**Comment by jluhrsen** [comment]
> > looks like the tests were running, but maybe the job timeout needs to be increased? aborted after the 4h timeout:
> > ```
> > INFO[2022-08-17T12:16:04Z] Running step e2e-network-mtu-migration-ovn-ipv6-baremetalds-e2e-test. 
> > {"component":"entrypoint","file":"k8s.io/test-infra/prow/entrypoint/run.go:165","func":"k8s.io/test-infra/prow/entrypoint.Options.ExecuteProcess","level":"error","msg":"Process did not finish before 4h0m0s timeout","severity":"error","time":"2022-08-17T13:05:45Z"}
> > ```
> 
> Is that something we can do per job? I though it was a hard timeout of 4h for all jobs.
> 
> **edit**: done, I didn't know we could do this.

cool. I would try to figure out how long some steps are taking and see if it makes sense. maybe it's not likely, but if you
were to find that some step (e.g., the normal e2e-test step) is taking abnormally long as compared to how long it takes
in a different job that's not doing mtu migration, then that would be worth investigating as a possible bug. If it's just that
all the new extra steps of doing the migration just add up to be longer than 4h, then so be it.

**Comment by jcaamano** [comment]
> > > looks like the tests were running, but maybe the job timeout needs to be increased? aborted after the 4h timeout:
> > > ```
> > > INFO[2022-08-17T12:16:04Z] Running step e2e-network-mtu-migration-ovn-ipv6-baremetalds-e2e-test. 
> > > {"component":"entrypoint","file":"k8s.io/test-infra/prow/entrypoint/run.go:165","func":"k8s.io/test-infra/prow/entrypoint.Options.ExecuteProcess","level":"error","msg":"Process did not finish before 4h0m0s timeout","severity":"error","time":"2022-08-17T13:05:45Z"}
> > > ```
> > 
> > 
> > Is that something we can do per job? I though it was a hard timeout of 4h for all jobs.
> > **edit**: done, I didn't know we could do this.
> 
> cool. I would try to figure out how long some steps are taking and see if it makes sense. maybe it's not likely, but if you were to find that some step (e.g., the normal e2e-test step) is taking abnormally long as compared to how long it takes in a different job that's not doing mtu migration, then that would be worth investigating as a possible bug. If it's just that all the new extra steps of doing the migration just add up to be longer than 4h, then so be it.

Yeah, I already did that exercise several times and it something really difficult to stay on top of. The job has several reboots and while I did change to reduce the time they take, things constantly change and so far it has a tendency of taking more time. We can definately take a further look into it but I think it is ok this jobs take a naturally long time given what they are testing.

Also I don't think we have any control over the timeout of pj-rehearse.

@jluhrsen quick! LGTM this before we have to deal with some other issue! :P

File: ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh
Comment (vrutkovs) [inline ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh:146]: Don't we have a test which ensures no important alerts are firing during the test?

**Review by vrutkovs — APPROVED** [review]
/approve
See comment above, but it can be clarified later

File: ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh
Comment (jcaamano) [reply ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh:146]: Hmm I didn't know that. Which period does it cover? The whole test run or only while e2e tests are running?

**Pull request merged**
